apps/web/src/state/farmsV4/state/poolApr/fetcher.ts

```diff
@@ -141,3 +182,168 @@ export const getAllNetworkMerklApr = async () => {
+    contracts: poolInfoCalls,
+    allowFailure: false,
+  })
+
+  return validPools.reduce((acc, pool, index) => {
+    const poolInfo = poolInfos[index]
+    if (!poolInfo) return acc
+    const key = `${chainId}:${safeGetAddress(pool.lpAddress)}`
+    return {
+      ...acc,
+      [key]: calcV3PoolApr({
+        pool,
+        cakePrice,
+        totalAllocPoint,
+        latestPeriodCakePerSecond,
+        poolInfo,
+      }),
+    }
+  }, {} as CakeApr)
+}
+
+const getV3PoolsCakeApr = async (pools: V3PoolInfo[]) => {
+  const cakePrice = await getCakePrice()
+  const poolsByChainId = groupBy(pools, 'chainId')
+  const aprs = await Promise.all(
+    Object.keys(poolsByChainId).map((chainId) =>
+      getV3PoolsCakeAprByChainId(poolsByChainId[Number(chainId)], Number(chainId), cakePrice),
+    ),
+  )
+  return aprs.reduce((acc, apr) => ({ ...acc, ...apr }), {})
+}
+
+const v3PoolCakeAprBatcher = create<CakeApr, V3PoolInfo, CakeApr>({
+  fetcher: getV3PoolsCakeApr,
+  resolver: (items, query) => {
+    const key = `${query.chainId}:${query.lpAddress}`
+    return { [key]: items[key] }
+  },
+  scheduler: windowedFiniteBatchScheduler({
+    windowMs: 60,
+    maxBatchSize: 100,
+  }),
+})
+
+const calcV2PoolApr = ({
+  pool,
+  cakePrice,
+  cakePerSecond,
+}: {
+  pool: V2PoolInfo | StablePoolInfo
+  cakePrice: BigNumber
+  cakePerSecond: bigint
+}) => {
+  const cakeOneYearUsd = cakePrice.times((cakePerSecond * BigInt(SECONDS_PER_YEAR)).toString()).dividedBy(1e18)
+
+  const baseApr = cakeOneYearUsd.dividedBy(pool.tvlUsd ?? 1)
+
+  return {
+    value: baseApr.toString() as `${number}`,
+    boost: baseApr.times(2.5).toString() as `${number}`,
+  }
+}
+
+const cakePriceCache = {
+  value: new BigNumber(0),
+  timestamp: 0,
+}
+const getCakePrice = async () => {
+  const now = Date.now()
+  // cache for 10 minutes
+  if (now - cakePriceCache.timestamp < 1000 * 60 * 10) {
+    return cakePriceCache.value
+  }
+  return new BigNumber(await getCakePriceFromOracle())
+}
+const getV2PoolsCakeAprByChainId = async (
+  pools: Array<V2PoolInfo | StablePoolInfo>,
+  chainId: number,
+  cakePrice: BigNumber,
+) => {
+  const client = publicClient({ chainId })
+  const validPools = pools.reduce((prev, pool) => {
+    if (pool.chainId !== chainId) return prev
+    const bCakeWrapperAddress = getUniversalBCakeWrapperForPool(pool)?.bCakeWrapperAddress
+    if (!bCakeWrapperAddress) return prev
+    return [...prev, { ...pool, bCakeWrapperAddress }]
+  }, [] as (PoolInfo & { bCakeWrapperAddress: Address })[])
+
+  const calls = validPools.map((pool) => {
+    return {
+      address: pool.bCakeWrapperAddress,
+      functionName: 'rewardPerSecond',
+      abi: v2BCakeWrapperABI,
+    } as const
+  })
+
+  const results = await client.multicall({
+    contracts: calls,
+    allowFailure: false,
+  })
+
+  return validPools.reduce((acc, pool, index) => {
+    const result = results[index]
+    if (!result) return acc
+    return {
```

Comment:
performance issue if validPools is large, can use `acc[`${chainId}:${safeGetAddress(pool.lpAddress)}`]=calcV2PoolApr(...)`

Reply:
have chunk the request, the max length is 100 now